pilot/pkg/model/validation.go

```diff
@@ -1547,27 +1547,42 @@ func checkServiceRoleBinding(in *rbac.ServiceRoleBinding) error {
 			errs = appendErrors(errs, fmt.Errorf("do not use * for names or not_names (in rule %d)", i))
 		}
 	}
-	if in.RoleRef != nil && len(in.Actions) > 0 {
-		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef` or `actions` can be specified"))
-		return errs
+	roleFieldCount := 0
+	if in.RoleRef != nil {
+		roleFieldCount++
+	}
+	if len(in.Actions) > 0 {
+		roleFieldCount++
 	}
-	if in.RoleRef == nil && len(in.Actions) == 0 {
-		errs = appendErrors(errs, fmt.Errorf("`roleRef` or `actions` must be specified"))
+	if in.Role != "" {
+		roleFieldCount++
+	}
+	if roleFieldCount != 1 {
+		errs = appendErrors(errs, fmt.Errorf("exactly one of `roleRef`, `role`, or `actions` must be specified"))
+		return errs
```

Comment:
Should we still continue validation to show all other potential problems? It was the pattern of the code is keep adding error to the return, right?

Reply:
This is actually better. Yes we should tell the user all the errors they have all at once. Updated. Thanks!